libr/arch/p/or1k/plugin.c

```diff
@@ -16,8 +16,22 @@ struct operands {
 };
 
 // XXX remove globals and move into init/fini data pointer or just reimplement this into an ctual anal plugin for 5.9 so maybe its not worth
-static R_TH_LOCAL ut32 cpu[32] = {0}; /* register contents */
-static R_TH_LOCAL ut32 cpu_enable; /* allows to treat only registers with known value as valid */
+struct or1k_regs {
+	ut32 cpu[32]; /* register contents */
+	ut32 cpu_enable; /* allows to treat only registers with known value as valid */
+};
+
+static bool or1k_init(RArchSession *s) {
+	r_return_val_if_fail (!s->user, false);
```

Comment:
this check seems wrong to me, s->user must be null because the user poitjner is not used yet. so better assert if the session is null instead of the user pointer

Reply:
I double check the r_return_val_if_fail macro, and I am pretty sure this will return only if `!(!s->user)`,  it asserts that `s == NULL`. I can also add `s != NULL` in the assert.

Reply:
Yes the assert wont fail. But at some point the arch api may allow to set a value in this user pointer and that will explode. Maybe we can even remove the field if we cant find an use for it in 5.9